### ISSUE TITLE
Sparse Adam with weight decay

### ISSUE THREAD

**Issue by magus96** [pytorch/pytorch#25484]
## Feature Request

Is it possible to implement sparse Adam with weight decay?

cc @vincentqb

**Comment by yf225** [comment]
@magus96 Yes it should be possible, and we would love contributions in this area if you are interested in implementing it.

cc @vincentqb

**Comment by NekuSakuraba** [comment]
I'd like to contribute with this implementation.
Can you give me a starting point?

**Comment by magus96** [comment]
Currently I'm just looking at the sparse adam which has already been implemented and making some minor changes and we can see where we go from there.

**Comment by smoothclear** [comment]
I'm also interested in this topic. Could I also join? @yf225 @NekuSakuraba @magus96

**Comment by magus96** [comment]
In fact I've already worked on it some. Check out the notebook linked and see if you can benchmark it.
